src/tlv/TlvCodec.ts

```diff
@@ -140,15 +141,15 @@ export class TlvCodec {
     }
 
     /** @see {@link MatterCoreSpecificationV1_0} § A.7 */
-    public static readTagType(reader: DataReaderLE): { tag: TlvTag, typeLength: TlvTypeLength } {
+    public static readTagType(reader: DataReader): { tag?: TlvTag, typeLength: TlvTypeLength } {
```

Comment:
Where does Little-Endian get specified now?

Reply:
I see the DataReader constructor has endian argument and defaults to little.

Reply:
Little-Endian is specified when the DataReader is constructed and there is no default.
I generalized DataReader to support LE and BE since node-matter uses both and matter.js will probably soon.

Looking at this code now, I realized that this change was not correct, this reader has to be LE for the codec to work, so generalizing it weaken this constraint, fixing this now by create again DataReaderLE extending DataReader